### ISSUE TITLE
aria-expanded support in Edge

### ISSUE THREAD

**Issue by LJWatson** [FreedomScientific/VFO-standards-support#43]
### Summary
Jaws does not indicate aria-expanded state consistently in Edge.

1. Open the [test case](https://test-cases.tink.uk/aria-expanded/index.html)
2. Use `b` to move Jaws' focus to the "Tequila" button
3. Note that Jaws does not announce the `aria-expanded` state
4. Use the arrow keys to move Jaws' focus away and back to the button
5. Note that Jaws does not announce the `aria-expanded` state
6. Use the tab key to move focus away from the button and back again
7. Note that Jaws announces the `aria-expanded` state
8. Use space or enter to activate the button
9. Note that Jaws continues to announce the `aria-expanded` state as "collapsed

### Expected result
* Jaws should indicate the `aria-expanded` state regardless of method used to focus on the button
* Jaws should indicate the "expanded" state when the value of `aria-expanded` is changed

### Actual result
See steps above.

### Example
https://test-cases.tink.uk/aria-expanded/index.html

#### JAWS version and build number
Jaws 2018.7012.10 

#### Operating System and version
Windows 10 64bit

#### Browser and version:
Edge 41

**Comment by devarshipant** [comment]
Leonie -- Both browser versions were consistent in conveying expanded [true / false] states in my testing. 

Besides what you found, I think there is a gap in how instructional cues are processed by JAWS 18 - IE 11 and JAWS 18 - Edge .

- JAWS 18 / IE11 / Win 10 / 64 bit: JAWS 18 speaks 'Tequila, use JAWS key + Alt + M to move to controlled element, expanded.' (_perhaps, aria-controls causes this bit to be spoken by JAWS 18_).

- JAWS 18 / Edge / Win 10 / 64 bit:  Does not give shortcut cues to get to controlled element, but indicates the use of spacebar to activate. 'Tequila button expanded, to activate press spacebar.'

**Comment by LJWatson** [comment]
@devarshipant
<Leonie -- Both browser versions were consistent in conveying expanded [true / false] states in my testing.<

Which browsers did you test in? The issue reported here is only with Edge.

Besides what you found, I think there is a gap in how instructional cues are processed by JAWS 18 - IE 11 and JAWS 18 - Edge .



That's quite possible. Could you file a sepearate issue that documents what you've found?

**Comment by devarshipant** [comment]
I tested in IE 11 and Edge. The version of Edge I have is 25.10586.672.0. What version of Edge did you test with?

**Comment by devarshipant** [comment]
Your post mentions Edge 41.

**Comment by LJWatson** [comment]
Edge 41 is the current version (released in September 2017). In Edge 25 was the collapsed/expanded state announced when you used the b shortcut key and/or the arrow keys to move focus to the button?

**Comment by devarshipant** [comment]
Expand / Collapse state was announced in Edge 25 when button had keyboard
focus.

On Jan 22, 2018 15:25, "Léonie Watson" <notifications@github.com> wrote:

Edge 41 is the current version (released in September 2017). In Edge 25 was
the collapsed/expanded state announced when you used the b shortcut key
and/or the arrow keys to move focus to the button?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/FreedomScientific/VFO-standards-support/issues/43#issuecomment-359553587>,
or mute the thread

**Comment by LJWatson** [comment]
Thanks for testing this. It sounds like we're experiencing the same thing in both Edge 25 and 41, but to confirm...

When you follow steps 2 and 4 (from the list in my first comment), does Jaws announce the state of the button in Edge 25?

When you follow step 6, does Jaws announce the state of the button in Edge 25?

**Comment by devarshipant** [comment]
Interesting, JAWS 18, at least for me, does not work in virtual cursor mode in Edge 25. I cannot use 'b' to move focus to the button. Insert Z toggles the cursor mode, but I cannot make it work in virtual cursor. It could be our enterprise settings. Here is the thing though -- in PC cursor, it does speak the state when I tab to it, and consistently (and correctly) speaks the updated state.

**Comment by stevefaulkner** [comment]
@devarshipant 
>Interesting, JAWS 18, at least for me, does not work in virtual cursor mode in Edge 25

Initial official support for Edge was not introduced in JAWS until [JAWS 2018](https://www.freedomscientific.com/downloads/JAWS/JAWSWhatsNew)

**Comment by devarshipant** [comment]
Thanks for the heads up @stevefaulkner. I will test with JAWS 2018 and share my findings.

**Comment by stevefaulkner** [comment]
Confirmed as JAWS 2018 bug, Edge exposes expanded state correctly in UIA, but it is not announced by JAWS

**Comment by ggordon-vfo** [comment]
Now that Microsoft has officially adopted Chromium as the platform for Edge, we're no longer making changes to Edge Classic. I'm therefore closing this issue.
